inc/bucket.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Each page has 50,000 / CORE_SITEMAPS_POSTS_PER_BUCKET buckets.
+ */
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * Register the Sitemap Bucket custom post-type.
+ */
+function core_sitemaps_bucket_register() {
+	$labels = array(
+		'name'          => _x( 'Sitemap Buckets', 'Sitemap Bucket General Name', 'core-sitemaps' ),
+		'singular_name' => _x( 'Sitemap Bucket', 'Sitemap Bucket Singular Name', 'core-sitemaps' ),
+	);
+	$args   = array(
+		'label'           => __( 'Sitemap Bucket', 'core-sitemaps' ),
+		'description'     => __( 'Bucket of sitemap links', 'core-sitemaps' ),
+		'labels'          => $labels,
+		'supports'        => array( 'editor', 'custom-fields' ),
+		'can_export'      => false,
+		'rewrite'         => false,
+		'capability_type' => 'post',
+	);
+	register_post_type( CORE_SITEMAPS_CPT_BUCKET, $args );
+}
+
+/**
+ * Calculate the sitemap bucket number the post belongs to.
+ *
+ * @param int $post_id Post ID.
+ *
+ * @return int Sitemap Page pagination number.
+ */
+function core_sitemaps_page_calculate_bucket_num( $post_id ) {
```

Comment:
I don't think we'll be able to rely on this approach for identifying which bucket a post belongs to. For example, imagine a site had 2000 `page` posts published before publishing a `post` post type. In that case, the first `post` would have an ID that is > 2000, but this would assume that the post should be located in the second bucket, which would be an incorrect assumption.

Reply:
Yup and I realised this close after submission, I should have remembered that each post type doesn't have its own ID numbering, instead of course there's just the single wp_posts table.